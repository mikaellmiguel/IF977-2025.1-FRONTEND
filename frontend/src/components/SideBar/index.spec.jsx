import React from "react";
import { render, screen } from "@testing-library/react";
import { SideBar } from ".";

describe("SideBar", () => {
  it("deve renderizar o logo e o nome da aplicação", () => {
    render(<SideBar />);
    expect(screen.getByAltText(/logo da aplicação/i)).toBeInTheDocument();
    expect(screen.getByText(/FiscalizaDeputado/i)).toBeInTheDocument();
  });

  it("deve renderizar todos os itens do menu principal", () => {
    render(<SideBar />);
    expect(screen.getByText(/Deputados/i)).toBeInTheDocument();
    expect(screen.getByText(/Pesquisar/i)).toBeInTheDocument();
    expect(screen.getByText(/Seguidos/i)).toBeInTheDocument();
    expect(screen.getByText(/Ranking de Gastos/i)).toBeInTheDocument();
  });

  it("deve renderizar todos os itens do menu inferior", () => {
    render(<SideBar />);
    expect(screen.getByText(/Perfil/i)).toBeInTheDocument();
    expect(screen.getByText(/Sair/i)).toBeInTheDocument();
  });

  it("deve renderizar todos os ícones esperados", () => {
    render(<SideBar />);
    // FaUserTie aparece duas vezes
    expect(document.querySelectorAll('svg').length).toBeGreaterThanOrEqual(5);
  });

  it("deve bater snapshot básico", () => {
    const { container } = render(<SideBar />);
    expect(container).toMatchSnapshot();
  });
});
