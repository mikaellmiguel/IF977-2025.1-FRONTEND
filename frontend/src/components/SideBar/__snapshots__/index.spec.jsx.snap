// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`SideBar deve bater snapshot básico 1`] = `
<div>
  <nav
    class="sc-bRKDuR hbJGRK"
  >
    <div
      class="sc-hvigdm jJlyzb"
    >
      <img
        alt="Logo da aplicação FiscalizaDeputado"
        src="test-file-stub"
      />
      <p>
        FiscalizaDeputado
      </p>
    </div>
    <div
      class="sc-fhHczv epXVLr"
    >
      <p>
        Deputados
      </p>
      <a
        class="sc-ggWZvA cmOylZ"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 448 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M224 256c70.7 0 128-57.3 128-128S294.7 0 224 0 96 57.3 96 128s57.3 128 128 128zm95.8 32.6L272 480l-32-136 32-56h-96l32 56-32 136-47.8-191.4C56.9 292 0 350.3 0 422.4V464c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48v-41.6c0-72.1-56.9-130.4-128.2-133.8z"
          />
        </svg>
         Pesquisar
      </a>
      <a
        class="sc-ggWZvA cmOylZ"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 16 16"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M9.828.722a.5.5 0 0 1 .354.146l4.95 4.95a.5.5 0 0 1 0 .707c-.48.48-1.072.588-1.503.588-.177 0-.335-.018-.46-.039l-3.134 3.134a6 6 0 0 1 .16 1.013c.046.702-.032 1.687-.72 2.375a.5.5 0 0 1-.707 0l-2.829-2.828-3.182 3.182c-.195.195-1.219.902-1.414.707s.512-1.22.707-1.414l3.182-3.182-2.828-2.829a.5.5 0 0 1 0-.707c.688-.688 1.673-.767 2.375-.72a6 6 0 0 1 1.013.16l3.134-3.133a3 3 0 0 1-.04-.461c0-.43.108-1.022.589-1.503a.5.5 0 0 1 .353-.146"
          />
        </svg>
         Seguidos
      </a>
      <a
        class="sc-ggWZvA cmOylZ"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 640 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M353.8 54.1L330.2 6.3c-3.9-8.3-16.1-8.6-20.4 0L286.2 54.1l-52.3 7.5c-9.3 1.4-13.3 12.9-6.4 19.8l38 37-9 52.1c-1.4 9.3 8.2 16.5 16.8 12.2l46.9-24.8 46.6 24.4c8.6 4.3 18.3-2.9 16.8-12.2l-9-52.1 38-36.6c6.8-6.8 2.9-18.3-6.4-19.8l-52.3-7.5zM256 256c-17.7 0-32 14.3-32 32l0 192c0 17.7 14.3 32 32 32l128 0c17.7 0 32-14.3 32-32l0-192c0-17.7-14.3-32-32-32l-128 0zM32 320c-17.7 0-32 14.3-32 32L0 480c0 17.7 14.3 32 32 32l128 0c17.7 0 32-14.3 32-32l0-128c0-17.7-14.3-32-32-32L32 320zm416 96l0 64c0 17.7 14.3 32 32 32l128 0c17.7 0 32-14.3 32-32l0-64c0-17.7-14.3-32-32-32l-128 0c-17.7 0-32 14.3-32 32z"
          />
        </svg>
         Ranking de Gastos
      </a>
    </div>
    <div
      class="sc-dTvVRJ keYwvr"
    >
      <a
        class="sc-ggWZvA cmOylZ"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 448 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M224 256c70.7 0 128-57.3 128-128S294.7 0 224 0 96 57.3 96 128s57.3 128 128 128zm95.8 32.6L272 480l-32-136 32-56h-96l32 56-32 136-47.8-191.4C56.9 292 0 350.3 0 422.4V464c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48v-41.6c0-72.1-56.9-130.4-128.2-133.8z"
          />
        </svg>
         Perfil
      </a>
      <a
        class="sc-ggWZvA cmOylZ"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 16 16"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M9.828.722a.5.5 0 0 1 .354.146l4.95 4.95a.5.5 0 0 1 0 .707c-.48.48-1.072.588-1.503.588-.177 0-.335-.018-.46-.039l-3.134 3.134a6 6 0 0 1 .16 1.013c.046.702-.032 1.687-.72 2.375a.5.5 0 0 1-.707 0l-2.829-2.828-3.182 3.182c-.195.195-1.219.902-1.414.707s.512-1.22.707-1.414l3.182-3.182-2.828-2.829a.5.5 0 0 1 0-.707c.688-.688 1.673-.767 2.375-.72a6 6 0 0 1 1.013.16l3.134-3.133a3 3 0 0 1-.04-.461c0-.43.108-1.022.589-1.503a.5.5 0 0 1 .353-.146"
          />
        </svg>
         Sair
      </a>
    </div>
  </nav>
</div>
`;
